internal/monitors/vsphere/metadata.yaml

```diff
@@ -0,0 +1,1093 @@
+        type: gauge
+        default: false
+        group: datastore
+      vsphere.datastore.read_kbs:
+        description: Rate of reading data from the datastore (kilobytes per second)
+        type: gauge
+        default: false
+        group: datastore
+      vsphere.datastore.size_normalized_datastore_latency_ms:
+        description: Storage I/O Control size-normalized I/O latency.
+        type: gauge
+        default: false
+        group: datastore
+      vsphere.datastore.total_read_latency_ms:
+        description: Average amount of time for a read operation from the vsphere.datastore.
+          Total latency = kernel latency + device latency.
+        type: gauge
+        default: false
+        group: datastore
+      vsphere.datastore.total_write_latency_ms:
+        description: Average amount of time for a write operation to the vsphere.datastore.
+          Total latency = kernel latency + device latency.
+        type: gauge
+        default: false
+        group: datastore
+      vsphere.datastore.write_kbs:
+        description: Rate of writing data to the datastore.
+        type: gauge
+        default: false
+        group: datastore
+      vsphere.disk.bus_resets:
+        description: Number of SCSI-bus reset commands issued during the collection interval.
+        type: counter
+        default: false
+        group: disk
+      vsphere.disk.commands:
+        description: Number of SCSI commands issued during the collection interval.
+        type: counter
+        default: false
+        group: disk
+      vsphere.disk.commands_aborted:
+        description: Number of SCSI commands aborted during the collection interval.
+        type: counter
+        default: false
+        group: disk
+      vsphere.disk.commands_averaged:
+        description: Average number of SCSI commands issued per second during the collection interval.
+        type: gauge
+        default: false
+        group: disk
+      vsphere.disk.device_latency_ms:
+        description: Average amount of time, in milliseconds, to complete a SCSI command from the physical device.
+        type: gauge
+        default: false
+        group: disk
+      vsphere.disk.device_read_latency_ms:
+        description: Average amount of time, in milliseconds, to read from the physical device.
+        type: gauge
+        default: false
+        group: disk
+      vsphere.disk.device_write_latency_ms:
+        description: Average amount of time, in milliseconds, to write to the physical device.
+        type: gauge
+        default: false
+        group: disk
+      vsphere.disk.kernel_latency_ms:
+        description: Average amount of time, in milliseconds, spent by VMkernel to process each SCSI command.
+        type: gauge
+        default: false
+        group: disk
+      vsphere.disk.kernel_read_latency_ms:
+        description: Average amount of time, in milliseconds, spent by VMkernel to process each SCSI read command.
+        type: gauge
```

Comment:
Should any of these millisecond metrics be gauges?  I'm confused why some are counters and some are gauges.

Reply:
I just went by the [vmware docs](https://code.vmware.com/apis/197/vsphere/doc/cpu_counters.html) here -- if it said it's a "delta" type, then I mapped that to a counter, otherwise a gauge. But this is just for the docs -- the actual type sent to the sfx api comes from this function in points.go:

```
func statsTypeToMetricType(statsType types.PerfStatsType) datapoint.MetricType {
	if statsType == types.PerfStatsTypeDelta {
		return datapoint.Counter
	}
	return datapoint.Gauge
}
```
It's always possible that I made a data entry error in the yaml though.